### PR TITLE
Rename Iceberg delete_orphan_files to remove_orphan_files

### PR DESCRIPTION
**Pull request by homar**
## Description
REMOVE_ORPHAN_FILES is a name used by iceberg and spark 
## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:
Renamed Iceberg DELETE_ORPHAN_FILES procedure to REMOVE_ORPHAN_FILES

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by findepi — APPROVED** [review]
This is a breaking change, but we have an unintentional inconsistency with Spark procedures in what was intended to be modeled after them.

**Review by hashhar — APPROVED** [review]
Unfortunate but it's better to do such changes sooner rather than later.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java [hunk 1]
```diff
@@ -1763,9 +1763,9 @@ public void testSparkReadsTrinoTableAfterOptimizeAndCleaningUp(StorageFormat sto
 
         onTrino().executeQuery(format("ALTER TABLE %s EXECUTE OPTIMIZE", trinoTableName));
         onTrino().executeQuery("SET SESSION iceberg.expire_snapshots_min_retention = '0s'");
-        onTrino().executeQuery("SET SESSION iceberg.delete_orphan_files_min_retention = '0s'");
+        onTrino().executeQuery("SET SESSION iceberg.remove_orphan_files_min_retention = '0s'");
         onTrino().executeQuery(format("ALTER TABLE %s EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')", trinoTableName));
-        onTrino().executeQuery(format("ALTER TABLE %s EXECUTE DELETE_ORPHAN_FILES (retention_threshold => '0s')", trinoTableName));
+        onTrino().executeQuery(format("ALTER TABLE %s EXECUTE REMOVE_ORPHAN_FILES (retention_threshold => '0s')", trinoTableName));
```
Comment (findinpath) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java:1768]: nit: Let's use the lowercased convention for procedure names (as documented in `iceberg.rst`) `REMOVE_ORPHAN_FILES` -> `remove_orphan_files`

Comment (homar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java:1768]: We do use VACUUM, OPTIMIZE, EXPIRE_SNAPSHOTS so I think REMOVE_ORPHAN_FIILES is perfectly fine here

Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java:1768]: makes sense, but orthogonal

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -221,20 +221,20 @@ otherwise the procedure will fail with similar message:
...
-``delete_orphan_files`` can be run as follows:
+``remove_orphan_files`` can be run as follows:
 
 .. code-block:: sql
 
-  ALTER TABLE test_table EXECUTE delete_orphan_files(retention_threshold => '7d')
+  ALTER TABLE test_table EXECUTE remove_orphan_files(retention_threshold => '7d')
 
-The value for ``retention_threshold`` must be higher than ``iceberg.delete_orphan_files.min-retention`` in the catalog
+The value for ``retention_threshold`` must be higher than ``iceberg.remove_orphan_files.min-retention`` in the catalog
```
[the remark this answers is not in the document]
Comment (findepi) [reply docs/src/main/sphinx/connector/iceberg.rst]: Thanks for fixing this, but keep separate commit

[the remark this answers is not in the document]
Comment (findepi) [reply docs/src/main/sphinx/connector/iceberg.rst]: a rename should be as mechanical as possible, and other fixes/improvements should be separate

**Review by findepi — APPROVED** [review]

**Comment by phd3** [comment]
LGTM, worth calling out the config property naming change in RN. 

posted on [slack](https://trinodb.slack.com/archives/CJ6UC075E/p1652977825795419), hopefully if anyone is seriously impacted we'll know if fallback is needed

**Comment by findepi** [comment]
> My one question would be if we should keep the old name as legacy config so this is not a breaking change..

The procedure name changes. 
While we could keep config name with `@LegacyConfig`, the damage will be done anyway. I think it's maybe better to break config name and procedure name alike.

**Comment by mosabua** [comment]
> > My one question would be if we should keep the old name as legacy config so this is not a breaking change..
> 
> The procedure name changes. While we could keep config name with `@LegacyConfig`, the damage will be done anyway. I think it's maybe better to break config name and procedure name alike.

Fair enough ... probably need to mention it as breaking change in the release notes..

**Comment by findepi** [comment]
Thanks everyone for the involvement.
@mosabua i am not adding RN for this, i think you'll do the wording better here. Thanks

**Pull request merged**